SimpleImageIO/FlipBook.cs

```diff
@@ -442,6 +473,110 @@ public GeneratedCode Generate() {
+
+        // Update image on Html
+        if (images.Count == 0)
+            throw new InvalidOperationException("No images in the flip book");
+
+        List<string> dataStrs = new();
+        List<string> typeStrs = new();
+        List<string> nameStrs = new();
+        int width = images[0].Image.Width;
+        int height = images[0].Image.Height;
+        
+        if (width != updateImage.Image.Width || height != updateImage.Image.Height)
+            throw new InvalidOperationException("Image resolutions differ");
+
+        // The RGBE only supports exactly 3 color channels, so we fall back to half
+        var targetType = updateImage.TargetType;
+        if (targetType == DataType.RGBE && updateImage.Image.NumChannels != 3)
+            targetType = DataType.Float16;
+
+        dataStrs.Add(targetType switch
+        {
+            DataType.Float32 => WriteImageAsFloat32(updateImage.Image),
+            DataType.RGBE => CompressImageAsRGBE(updateImage.Image),
+            DataType.PNG => CompressImageAsPNG(updateImage.Image),
+            DataType.Float16 => WriteImageAsFloat16(updateImage.Image),
+            DataType quality => CompressImageAsJPEG(updateImage.Image, (int)quality)
+        });
+
+        typeStrs.Add(targetType switch
+        {
+            DataType.Float32 => "single",
+            DataType.RGBE => "rgbe",
+            DataType.Float16 => "half",
+            _ => "ldr"
+        });
+
+        nameStrs.Add(updateImage.Name);
+
+        // string id = "flipbook-" + Guid.NewGuid().ToString();
+        string id = this.containerId;
+        string html = $"<div id='{id}' style='width:{htmlWidth}px; height:{htmlHeight}px; resize: both; overflow: auto;'></div>";
+
+        string initialTMOStr = "null";
+        if (initialTMO != null) {
+            initialTMOStr = JsonSerializer.Serialize(initialTMO);
+        }
+
+        List<string> tmoOverrideStrs = [];
+        foreach (var img in images) {
+            if (img.TMOOverride == null)
+                tmoOverrideStrs.Add("null");
+            else
+                tmoOverrideStrs.Add(JsonSerializer.Serialize(img.TMOOverride));
+        }
+
+        string json = $$"""
+        {
+            "width": {{width}},
+            "height": {{height}},
+            "containerId": "{{id}}",
+            "key": {{JsonSerializer.Serialize(key)}},
+            "initialZoom": {{initialZoom.ToString()}},
+            "initialTMO": {{initialTMOStr}},
+            "initialTMOOverrides": [
+                {{string.Join(',', tmoOverrideStrs)}}
+            ],
+            "names": [{{string.Join(',', nameStrs.Select(n => $"\"{n}\""))}}],
+            "dataUrls": [{{string.Join(',', dataStrs.Select(n => $"\"{n}\""))}}],
+            "types": [{{string.Join(',', typeStrs.Select(n => $"\"{n}\""))}}],
+            "colorTheme": "{{theme}}",
+            "groupName": "{{groupName}}",
+            "hideTools": {{hideTools.ToString().ToLowerInvariant()}}
+        }
+        """;
```

Comment:
You should try and remove the code duplication with `Generate` here, will be annoying to update in the future otherwise.